web/source/kmwkeyboards.ts

```diff
@@ -0,0 +1,1383 @@
+      +'?jsonp=keyman.register';  
+
+    var kbdManager = this;
+    
+    // Set callback timer
+    tFlag='&timerid='+window.setTimeout(
+      function(){
+        kbdManager.serverUnavailable(cmd);
+      }
+      ,10000);    
+  
+    Lscript.charset="UTF-8";                     
+    Lscript.src = URL+cmd+tFlag;       
+    Lscript.type = 'text/javascript';       
+    try {                                  
+      document.body.appendChild(Lscript);  
+      }
+    catch(ex) {                                                     
+      document.getElementsByTagName('head')[0].appendChild(Lscript);
+      }                  
+  }
+
+  /**
+   *  Display warning if Keyman Cloud server fails to respond
+   * 
+   *  @param  {string}  cmd command string sent to Cloud
+   *          
+   **/       
+  private serverUnavailable(cmd) {
+    this.keymanweb.util.alert(cmd == '' ? 'Unable to connect to Keyman Cloud server!' : cmd);
+    this.keymanweb.warned=true;
+  }
+
+  /**
+   * Build 362: removeKeyboards() remove keyboard from list of available keyboards
+   * 
+   * @param {string} x keyboard name string
+   * 
+   */  
+  removeKeyboards(x: string) {
+    if(arguments.length == 0) {
+      return false;
+    }
+
+    var i, j;
+    var success = true, activeRemoved = false, anyRemoved = false;;
+
+    for(i=0; i<arguments.length; i++) {           
+      for(j=this.keyboardStubs.length-1; j>=0; j--) {
+        if('Keyboard_'+arguments[i] == this.keyboardStubs[j]['KI'] && this.keyboardStubs.length > 1) {                 
+          if('Keyboard_'+arguments[i] == this.getActiveKeyboardName()) {
+            activeRemoved = true;
+          }
+
+          anyRemoved = true;
+          this.keyboardStubs.splice(j,1);
+          break;
+        }
+      }
+
+      if(j < 0) {
+        success = false;
+      }
+    } 
+
+    if(activeRemoved) {
+      // Always reset to the first remaining keyboard
+      this._SetActiveKeyboard(this.keyboardStubs[0]['KI'],this.keyboardStubs[0]['KLC'],true);
+      // This is likely to be triggered by a UI call of some sort, and we need to treat
+      // this call as such to properly maintain the globalKeyboard setting.
+      this.keymanweb._JustActivatedKeymanWebUI = 1;
+    }
+
+    if(anyRemoved) {
+      // Update the UI keyboard menu
+      this.doKeyboardUnregistered();
+    }
+      
+    return success;
+  }
+
+  /**
+   * Function     registerKeyboard  KR                    
+   * Scope        Public
+   * @param       {Object}      Pk      Keyboard  object
+   * Description  Register and load the keyboard
+   */    
+  _RegisterKeyboard(Pk) {
+    // If initialization not yet complete, list the keyboard to be registered on completion of initialization
+    if(!this.keymanweb.initialized) {
+      this.deferredKR.push(Pk);
+      return;
+    }
+    
+    var Li,Lstub;
+
+    // For package namespacing with KMEA/KMEI.
+    if(this.keymanweb.isEmbedded) {
```

Comment:
Is it possible to extract this code into **kmwembedded.js** so that we can keep embedded-specific code out of the distributed kmw build?